### ISSUE TITLE
Uncaught ReferenceError: noral is not defined

### ISSUE THREAD

**Issue by vaibhavsagar** [leaningtech/webvm#103]
I tried to fork this repo [here](https://github.com/vaibhavsagar/webvm) and deploy using [this Dockerfile](https://github.com/vaibhavsagar/webvm/blob/main/dockerfiles/ghc) but when I visit the webpage I get this error in the JavaScript console:

```
Uncaught ReferenceError: noral is not defined
    getSharedArrayBufferMissingMessage http://vaibhavsagar.com/webvm/:166
    runBash http://vaibhavsagar.com/webvm/:281
    loadCX http://vaibhavsagar.com/webvm/:352
    async* http://vaibhavsagar.com/webvm/:355
webvm:166:120
    getSharedArrayBufferMissingMessage http://vaibhavsagar.com/webvm/:166
    runBash http://vaibhavsagar.com/webvm/:281
    (Async: EventListener.handleEvent)
    loadCX http://vaibhavsagar.com/webvm/:352
    InterpretGeneratorResume self-hosted:1412
    AsyncFunctionNext self-hosted:799
    (Async: async)
    <anonymous> http://vaibhavsagar.com/webvm/:355
```

Is this something I'm doing wrong somewhere?

**Comment by alexp-sssup** [comment]
There was a typo in the handling on error messages, now fixed, please update your fork.

Please notice that the error happens since SharedArrayBuffer is not enabled. You need to supply the right header for that to work. When deployed to GitHub pages WebVM is integrated with a service worker to automatically supply the right headers. I would recommend to follow the README instructions and deploy to GitHub Pages if you don't have previous experience with using SharedArrayBuffer and Cross Origin Isolation

**Comment by vaibhavsagar** [comment]
I was using GitHub Pages already, but it turned out I needed to enable "Enforce HTTPS" in my Pages settings before this would work (in Firefox).

**Comment by alexp-sssup** [comment]
That's correct. Cross origin isolation can only work in secure contexts.

**Comment by vaibhavsagar** [comment]
Maybe you could add this to your instructions?

**Comment by alexp-sssup** [comment]
The requirements for SharedArrayBuffer are well documented: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer

**Comment by vaibhavsagar** [comment]
Yes, but this means your "Fork, deploy, customize" instructions are incomplete. I created a PR: https://github.com/leaningtech/webvm/pull/105.
